### PR TITLE
Add job to build on QEMU with the right kernel to be able to use io_u…

### PR DESCRIPTION
**Pull request by normanmaurer**
…ring

Motivation:

Add scripts and job config which will spin up a QEMU VM and run the build on it. This allows us to run all io_uring tests as we can use the right kernel

Modifications:

- Add scripts
- Add config
- Add job that builds via QEMU

Result:

Run unit tests during PR validation

### CODE REVIEW

File: .github/config/user-data [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+     netty:netty
+  expire: False
+
+# Other settings
+resize_rootfs: True
+ssh_pwauth: True
+timezone: Europe/Berlin
+
+packages:
+ - java-11-openjdk-devel
```
Comment (1Jo1) [inline .github/config/user-data:19]: it would be nice test it on different java versions? or what do you think?

**Review by 1Jo1 — COMMENTED** [review]
@normanmaurer looks good 👍  I am not sure if i have understood this correctly, the goal is to build a kernel image(only the current kernel version?) and test netty in qemu? 
I guess it seems to be difficult to boot a custom kernel with a fedora cloud image(cloud init) right as you need specify the root partition name or UUID

**Comment by normanmaurer** [comment]
@1Jo1 actually we don't need a custom kernel... the kernel that is used is good enough for now.

File: .github/config/user-data (hunk 1 above)
Comment (normanmaurer) [reply .github/config/user-data:19]: I would prefer to keep it with one version for now as its sloooooow

File: .github/scripts/build_kernel.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+if [ "$#" -ne 1 ]; then
+    echo "Expected kernel image as argument"
+    exit 1
+fi
+
+KERNEL=linux-5.9.16
+wget -q https://cdn.kernel.org/pub/linux/kernel/v5.x/$KERNEL.tar.xz
+tar xf $KERNEL.tar.xz
+cd $KERNEL && cp /boot/config-`uname -r`* .config
```
[the remark this answers is not in the document]
Comment (normanmaurer) [reply .github/scripts/build_kernel.sh:11]: like I said we don't need this anymore atm

**Comment by 1Jo1** [comment]
@normanmaurer ok I got it, qemu tests are really slow here...as github action servers has only 2 cores...what about using github self hosted runners?

**Comment by normanmaurer** [comment]
@1Jo1 as stated in the docs of GitHub Actions you should not use self hosted runners for public projects as it may be a security problem

**Comment by 1Jo1** [comment]
@normanmaurer yes only if your workflows is not triggered by pull request event(according to github docs), we "could" use self hosted runners which are triggered by master push event

**Comment by 1Jo1** [comment]
we could use that https://github.com/marketplace/actions/publish-unit-test-results instead using a script?

**Comment by normanmaurer** [comment]
@1Jo1 let me start with the script and then check the action. We may want to use this everywhere.

**Pull request merged**